### PR TITLE
Add lint prefer_double_quotes

### LINKED ISSUE #1096: Add prefer_double_quotes linter
For those of us using languages with double-quotes (e.g. C#) on the back-end, and using Dart on the front-end, it's quite annoying to have to switch between two quote demarcations. There already exists a `prefer_single_quotes` linter, but there is no counterpart for double quotes. This is a request to add one, which should be fairly straightforward given the existence of the `prefer_single_quotes` linter.

### PR DESCRIPTION
**Pull request by a14n**
Fixes #1096

### CODE REVIEW

**Comment by pq** [comment]
It looks like this is failing the integration tests, since `p5.dart` contains single quotes:

```dart
import 'package:p6/p6_lib.dart';
```

Maybe this is as simple as adding an analysis_options file in `_data/p5`?

File: lib/src/rules/prefer_single_quotes.dart [hunk 1]
```diff
@@ -117,7 +122,8 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   @override
   void visitSimpleStringLiteral(SimpleStringLiteral string) {
-    if (string.isSingleQuoted || string.value.contains("'")) {
+    if (useSingle && (string.isSingleQuoted || string.value.contains("'")) ||
```
Comment (pq) [inline lib/src/rules/prefer_single_quotes.dart]: Perhaps extract a function that can be used here and below?

**Comment by kevmoo** [comment]
No! Don't let folks go the other way ❗️

**Comment by kevmoo** [comment]
...I'm kinda joking...but also kinda serious. It's great if push the community in one direction. Who is asking for this?

**Comment by bwilkerson** [comment]
See #1096.

**Comment by a14n** [comment]
>It looks like this is failing the integration tests, since p5.dart contains single quotes

I really don't understand why it fails because there's no place where this new `prefer_double_quotes` is used :-/

**Comment by pq** [comment]
Yes. Right.  That's a good point... I'm unsure now but I'm wondering if, in the absence of an options file, the linter CLI isn't defaulting to enabling all the lints? 😬

File: lib/src/rules/prefer_single_quotes.dart (hunk 1 above)
Comment (a14n) [reply lib/src/rules/prefer_single_quotes.dart]: The conditions are not exactly the same.

File: test/integration_test.dart [hunk 2]
```diff
@@ -105,8 +105,9 @@ defineTests() {
       group('.packages', () {
         test('basic', () async {
           // Requires .packages to analyze cleanly.
-          await cli
-              .run(['test/_data/p5', '--packages', 'test/_data/p5/_packages']);
+          await cli.runLinter(
+              ['test/_data/p5', '--packages', 'test/_data/p5/_packages'],
+              LinterOptions([]));
```
Comment (pq) [inline test/integration_test.dart]: 👍

**Pull request merged**